update_bash_env.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""
+Generate a file for use by BASH_ENV
+"""
+
+import json
+from pathlib import Path
+from logging import getLogger, basicConfig
+from argparse import ArgumentParser
+from yaml import safe_load, YAMLError
+from jinja2 import Environment, FileSystemLoader
+
+formatting = json.dumps(
+    {
+        "timestamp": "%(asctime)s",
+        "namespace": "%(name)s",
+        "loglevel": "%(levelname)s",
+        "message": "%(message)s",
+    }
+)
+basicConfig(level="WARNING", format=formatting)
+LOG = getLogger("easy_infra")
+
+
+def render(*, template_file: Path, config: dict, output_file: Path) -> None:
+    """Render the functions file"""
+    folder = str(template_file.parent)
+    file = str(template_file.name)
+    template = Environment(loader=FileSystemLoader(folder)).get_template(file)
+    out = template.render(config)
+    output_file.write_text(out)
+    output_file.chmod(0o755)
+
+
+def create_arg_parser() -> ArgumentParser:
+    """Parse the arguments"""
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--config-file",
+        type=lambda p: Path(p).absolute(),
+        default=Path("easy_infra.yml").absolute(),
+        help="specify a config file",
+    )
+    parser.add_argument(
+        "--output",
+        type=lambda p: Path(p).absolute(),
+        default=Path("functions").absolute(),
+        help="specify an output file",
+    )
+    parser.add_argument(
+        "--template-file",
+        type=lambda p: Path(p).absolute(),
+        default=Path("functions.j2").absolute(),
+        help="specify a jinja2 template",
+    )
+    return parser
+
+
+def get_args_config() -> dict:
+    """Get the configs passed as arguments"""
+    parser = create_arg_parser()
+    parsed_args = vars(parser.parse_args())
+    return parsed_args
+
+
+def parse_file_config(*, config_file: Path) -> dict:
+    """Parse the easy_infra config file"""
+    # Filter
+    suffix_whitelist = {".yml", ".yaml"}
+
+    if config_file.suffix not in suffix_whitelist:
+        LOG.error("Suffix for the config file %s is not allowed", config_file)
+        raise RuntimeError
+
+    try:
```

Comment:
do you log the error somewhere else later? if not, i'm not sure I would eat the exception, I would want the information

Reply:
Not sure what you mean, that's what `raise x from y` does

Reply:
you aren't logging the error object

Reply:
LOG.error(f"Suffix for the configuration file {config_file} is not allowed", e)

Reply:
Right, I am intentionally leaving `exc_info=True` out because I still raise the exception, including the cause, and the log is more user friendly.

Reply:
ok, but if this goes wrong, I wouldn't want to have to figure out what the hell is going on

Reply:
It would print the exception chain to stdout still.

Reply:
yes, I would want that.

How about adding a verbose -v flag and outputting the error to the log if set?

Reply:
as a follow on I guess

Reply:
How about 3d17393abd889e04af7a8fbe003cd68b470f2ca9